### PR TITLE
Hw02 exppress

### PR DESCRIPTION
**Pull request by DorotaHeczko**

### CODE REVIEW

File: models/contacts.js [hunk 1]
```diff
@@ -1,19 +1,117 @@
...
+    console.log(error);
+    throw error; 
+  }
+};
+
+
+const removeContact = async (contactId) => {
+  try {
+    const contacts = await getContacts();
+    const targetContact = contacts.find((data) => data.id === contactId);
```
Comment (AdamChodowiecGoIt91) [inline models/contacts.js]: `data` has no specific meaning it can be everything. `contact` would be better.

File: models/contacts.js [hunk 2]
```diff
@@ -1,19 +1,117 @@
...
+    throw error; 
+  }
+};
+
+
+const getContactById = async (contactId) => {
+  try {
+    const contacts = await getContacts();
+
+    return contacts.find((data) => data.id === contactId);
```
Comment (AdamChodowiecGoIt91) [inline models/contacts.js]: `data` has no specific meaning it can be everything. `contact` would be better.

File: routes/api/contacts.js [hunk 3]
```diff
@@ -1,25 +1,90 @@
...
+router.delete("/:contactId", async (req, res, next) => {
+  try {
+    const { contactId } = req.params;
+    const isContactDeleted = await contacts.removeContact(contactId);
+
+    isContactDeleted
+      ? res.status(200).json({ message: "contact deleted" })
+      : res.status(404).json({ message: "not found" });
+  } catch (error) {
+    console.log(error);
```
Comment (AdamChodowiecGoIt91) [inline routes/api/contacts.js]: what user will receive as response if you land in `catch`?

File: routes/api/contacts.js [hunk 4]
```diff
@@ -1,25 +1,90 @@
...
+    if (error) return res.status(400).send({ message: error.details });
 
-module.exports = router
+    const allContactsData = await contacts.addContact(body);
+    res.status(201).json({
+      message: "contact added",
+      data: { allContactsData },
+    });
+  } catch (error) {
+    console.log(error);
```
Comment (AdamChodowiecGoIt91) [inline routes/api/contacts.js:50]: what user will receive as response if you land in `catch`?

File: routes/api/contacts.js [hunk 5]
```diff
@@ -1,25 +1,90 @@
...
+    if (modifiedContact) {
+      res.status(200).json({
+        message: "contact edited",
+        data: { modifiedContact },
+      });
+    } else {
+      res.status(404).json({ message: "Not found" });
+    }
+  } catch (error) {
+    console.log(error);
```
Comment (AdamChodowiecGoIt91) [inline routes/api/contacts.js:90]: what user will receive as response if you land in `catch`?

File: models/contacts.js (hunk 1 above)
Comment (DorotaHeczko) [reply models/contacts.js]: done

File: models/contacts.js (hunk 2 above)
Comment (DorotaHeczko) [reply models/contacts.js]: done

File: routes/api/contacts.js (hunk 3 above)
Comment (DorotaHeczko) [reply routes/api/contacts.js]: added answer

File: routes/api/contacts.js (hunk 4 above)
Comment (DorotaHeczko) [reply routes/api/contacts.js:50]: added answer

File: routes/api/contacts.js (hunk 5 above)
Comment (DorotaHeczko) [reply routes/api/contacts.js:90]: added answer

**Pull request merged**